### PR TITLE
Add prevent_forking_outside_group

### LINKED ISSUE #529: Add "prevent forking project outside of group" attribute to "gitlab_group" resource
As the title states there is an option in GitLab to prevent the forking of projects outside of the group that owns them.  
The gitlab_group resouce would have to be modified to add this in.

applicable license tier's
- premium
- silver

Relevant Links:
- https://docs.gitlab.com/ee/user/group/#prevent-project-forking-outside-group

### PR DESCRIPTION
**Pull request by jtymes**
## Description

Closes #529 

### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

File: internal/provider/data_source_gitlab_group.go
Comment (timofurrer) [inline internal/provider/data_source_gitlab_group.go:95]: NIT:

```suggestion
				Description: "When enabled, users can not fork projects from this group to external namespaces.",
```

File: internal/provider/resource_gitlab_group.go
Comment (timofurrer) [inline internal/provider/resource_gitlab_group.go:152]: NIT:

```suggestion
				Description: "When enabled, users can not fork projects from this group to external namespaces.",
```

File: internal/provider/resource_gitlab_group.go
Comment (timofurrer) [inline internal/provider/resource_gitlab_group.go:240]: NIT:

```suggestion
			return diag.Errorf("could not update group after creation %q: %s", d.Id(), err)
```

**Review by timofurrer — CHANGES_REQUESTED** [review]
I have a few minimal nitpicks, but otherwise LGTM 🎉

File: internal/provider/resource_gitlab_group_test.go
Comment (timofurrer) [inline internal/provider/resource_gitlab_group_test.go:263]: Just noticed this now, but couldn't you use `resource.TestCheckResourceAttrPtr` instead of the custom check function?

In addition, it would be nice to test this fields update logic.

**Review by timofurrer — CHANGES_REQUESTED** [review]

File: internal/provider/resource_gitlab_group_test.go
Comment (jtymes) [reply internal/provider/resource_gitlab_group_test.go:263]: Thank you for the pointer. I'm still new to the provider SDK, I guess I should take a look at the tools available to me 😄 .

**Review by timofurrer — APPROVED** [review]

**Pull request merged**
